src/components/Microsoft.DotNet.UpgradeAssistant/UpgraderManager.cs

```diff
@@ -40,11 +40,9 @@ public async Task<IEnumerable<UpgradeStep>> InitializeAsync(IUpgradeContext cont
                 await _restorer.RestorePackagesAsync(context, context.EntryPoint, token).ConfigureAwait(false);
             }
 
-            return GetStepsForContext(context);
+            return AllSteps;
```

Comment:
Why return `AllSteps` and then filter in `ConsoleUpgrade.cs` by applicable? Why not keep the filter here?

Reply:
I debated this. I thought it was better to filter in ConsoleUpgrade for a couple reasons:

1. ConsoleUpgrade has should filter anyhow to make sure that it's also filtering out sub-steps that don't apply.
2. Even though we don't show inapplicable steps, I feel like that's a UI/UX decision and there could conceivably be a front-end someday that wants to list things like that grayed out or something. So I figured it made more sense to pass the whole list to the front-end and let it make the decision about how to display things.

Reply:
Cool!